evm_loader/program/src/executor_state.rs

```diff
@@ -765,8 +765,8 @@ pub struct ExecutorState<'a, B: AccountStorage> {
 impl<'a, B: AccountStorage> ExecutorState<'a, B> {
     #[must_use]
     #[allow(clippy::unused_self)]
-    pub fn block_hash(&self, _number: U256) -> H256 {
-        H256::default()
+    pub fn block_hash(&self, number: U256) -> H256 {
+        self.backend.block_hash(number)
```

Comment:
block_hash need to be cached. So we have the same value in different iterations.

Reply:
Done